modules/plugins/languages/haskell.nix

```
@@ -0,0 +1,99 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  inherit (builtins) isList;
+  inherit (lib.types) either package listOf str;
+  inherit (lib.options) mkEnableOption mkOption;
+  inherit (lib.strings) optionalString;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.nvim.types) mkGrammarOption;
+  inherit (lib.nvim.dag) entryAnywhere;
+  inherit (lib.nvim.lua) expToLua;
+  inherit (pkgs) haskellPackages;
+  cfg = config.vim.languages.haskell;
+in {
+  options.vim.languages.haskell = {
+    enable = mkEnableOption "Haskell support";
+    treesitter = {
+      enable = mkEnableOption "Treesitter support for Haskell" // {default = config.vim.languages.enableTreesitter;};
+      package = mkGrammarOption pkgs "haskell";
+    };
+
+    lsp = {
+      enable = mkEnableOption "LSP support for Haskell" // {default = config.vim.languages.enableLSP;};
+      package = mkOption {
+        description = "Haskell LSP package or command to run the Haskell LSP";
+        example = ''[ (lib.getExe pkgs.haskellPackages.haskell-language-server) "--debug" ]'';
+        default = haskellPackages.haskell-language-server;
+        type = either package (listOf str);
+      };
+    };
+
+    dap = {
+      enable = mkEnableOption "DAP support for Haskell" // {default = config.vim.languages.enableDAP;};
+      package = mkOption {
+        description = "Haskell DAP package or command to run the Haskell DAP";
+        default = haskellPackages.haskell-debug-adapter;
+        type = either package (listOf str);
+      };
+    };
+  };
+  config = mkIf cfg.enable (mkMerge [
+    (mkIf cfg.treesitter.enable {
+      vim.treesitter = {
+        enable = true;
+        grammars = [cfg.treesitter.package];
+      };
+    })
+    (mkIf (cfg.dap.enable || cfg.lsp.enable) {
+      vim = {
+        startPlugins = ["haskell-tools-nvim"];
+        luaConfigRC.haskell-tools-nvim =
+          entryAnywhere
```

This does not guarantee that this code is run after `default_on_attach` has been defined.

```suggestion
        pluginRC.haskell-tools-nvim =
          entryAfter
          ["lsp-setup"]
```
If you don't mind, you could also fix this in the Rust module, as this mistake was also made there.

---

Fixed both.